osu.Game.Rulesets.Catch/Difficulty/Skills/Movement.cs

```diff
@@ -26,10 +27,20 @@ public class Movement : Skill
         private float lastDistanceMoved;
         private double lastStrainTime;
 
+        /// <summary>
+        /// The speed multiplier applied to the player's catcher.
+        /// </summary>
+        private readonly double catcherSpeedMultiplier;
+
         public Movement(Mod[] mods, float halfCatcherWidth)
             : base(mods)
         {
             HalfCatcherWidth = halfCatcherWidth;
+
+            // In catch, rate adjustment mods do not only affect the timings of hitobjects,
+            // but also the speed of the player's catcher, which has an impact on difficulty
+            var rateAdjustMod = mods.FirstOrDefault(m => m is ModRateAdjust);
```

Comment:
Is there a reason for calculating it here instead of passing it in from `DifficultyCalculator`? It would need `CreateSkills()` to take in the clock rate, but I think that's fine.

In the future we'll probably change that to be a custom object instead of a simple `double`, that you can then query the rate from via `ClockRateAt(double time)`.

Reply:
Yea that's totally possible too.
I merely did it this way because clockrate can be easily derived from mods, so it wouldn't require any additional parameters to be passed, but if the plan is to replace clockrate with a custom object capable of describing variable clockrate, then that might be the better approach.

Reply:
Sounds fine.

Reply:
Changed in f51413ead9a0c1afba047526ff9fae183ff1c9db